client_cli/src/main.rs

```diff
@@ -1023,3 +1118,80 @@ mod json {
         }
     }
 }
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use serde_json::Value as JsonValue;
+
+    use super::*;
+
+    #[test]
+    fn parse_ipv4_address() {
+        let actual = ValueArg::from_str("192.168.0.1").expect("Failed to parse valid IPv4 address");
+        assert_eq!(
+            actual,
+            ValueArg(Value::Ipv4Addr(Ipv4Addr::new([192, 168, 0, 1])))
+        );
+    }
```

Comment:
I would suggest to organise all "success" cases into a single test with reduced repetition using a macro. For example:

```rust
#[test]
fn parse_value_arg_cases() {
    macro_rules! case {
        ($input:expr, $expected:expr) => {
            let ValueArg(actual) =
                ValueArg::from_str($input).expect("should not fail with valid input");
            assert_eq!(actual, $expected);
        };
    }

    case!(
        "192.168.0.1",
        Value::Ipv4Addr(Ipv4Addr::new([192, 168, 0, 1]))
    );
    case!("true", Value::Bool(true));
    case!("false", Value::Bool(false));
    // ...
}
```

This way, it is easy to write many-many cases and see them all in a single place, which helps a lot for review and future maintenance.

You wrote many test cases already, but they are not really comprehensive, e.g. it is not obvious what numeric suffixes are supported.